lib/enigma/encryption.livemd

```diff
@@ -297,6 +297,29 @@ graph TD;
   rk <--encrypted room approval----> r
 ```
 
+## Shamir's Secret Sharing
+
+If we need to protect our secret, we could use secret sharing.
+
+<ul>
+  This is how it works: 
+  <li>
+    For starters, we take 3 parameters: our secret, a total amount of shares we need to generate, and a threshold, that indicates how many shares would be enough to recover our secret down the line;
+  </li>
+  <li>
+    Imagine that our secret is represented by a number. This number could be placed on the Oy axis. If we wanted to hide that point, we could represent it via a number of other points that would form a line, that intercepts Oy axis. The complexity of that line determines how easy it would be to calculate, which function represents that line, and by proxy, our secret. For instance, if our line was straight, 2 points would be enough to know where our secret lies on Oy axis;
+  </li>
+  <li>
+    Needless to say that the threshold shouldn't be bigger than a total amount of shares, you can't really recover your secret if you don't have enough points to form a needed line;
+  </li>
+  <li>
+    And that's basically it, the output of the share generation should be a list of shares, which we can handle out each to different users;
+  </li>
+  <li>
+    Once we need to recover our secret, we can ask these users what share they have, and after getting enough of them, our secret should be safely recovered.
+  </li>
+</ul>
```

Comment:
1. use markdown here
2. this document is more about data flow. A diagram with secrets splitted into many shares and recovering from 3 of them is better here